remotespark/wrapperkernel/codetransformers.py

```diff
@@ -0,0 +1,159 @@
+# Copyright (c) 2015  aggftw@gmail.com
+# Distributed under the terms of the Modified BSD License.
+
+from remotespark.utils.constants import Constants
+
+
+class UserCodeTransformerBase(object):
+    def __init__(self, subcommand):
+        self.name = subcommand
+
+    def get_code_to_execute(self, session_started, connection_string, force, output_var, command):
+        raise NotImplementedError()
+
+
+class NotSupportedTransformer(UserCodeTransformerBase):
+    def get_code_to_execute(self, session_started, connection_string, force, output_var, command):
+        error_to_show = "Magic '{}' not supported.".format(self.name)
+        code_to_run = ""
+        begin_action = Constants.do_nothing_action
```

Comment:
I don't know if I'm crazy about using string-based "indirection" for actions -- it might be simpler to just use functions.  I could go either way on that.